File: nodes/string-split/string-split.process.py
async def process(inputs, settings, config, nodeConfig):
  text = str(inputs.get("text", ""))
  delimiter = settings.get("delimiter", " ")
  limit = int(settings.get("limit", -1))
  
  if limit > 0:
    parts = text.split(delimiter, limit)
    parts = parts[:limit]
  else:
    parts = text.split(delimiter)
    
  return {
    "parts": parts
  } 

File: sdks/nodejs/nodes/string-split/string-split.process.js
export default async ({inputs, settings, config}) => {

  const text = String(inputs.text || "");
  const delimiter = settings.delimiter || " ";
  const limit = parseInt(settings.limit) || -1;
  
  const parts = limit > 0 ? 
    text.split(delimiter, limit) : 
    text.split(delimiter);
  
  return { parts };
}; 